File: TP1 - Dart/domain/entities/user_permissions.dart
class UserPermissions {
  final String _id;
  List<String> _permissions;

  UserPermissions(this._id) : _permissions = [''];

  UserPermissions.generalUser(this._id) : _permissions = ['schedule'];

  UserPermissions.administrativeAssistantUser(this._id) : _permissions = ['schedule'];

  UserPermissions.administrativeUser(this._id) : _permissions = ['schedule'];

  // Getter's
  String get id => this._id;
  List<String> get permissions => this._permissions;

  // Setter's
  set permissions(List<String> permissions) => this._permissions = permissions;
}

File: TP2 - Java/src/domain/entities/UserPermissions.java
package domain.entities;

import java.util.List;

public class UserPermissions {
    private final String id;
    private List<String> permissions;

    public UserPermissions(String id, List<String> permissions) {
        this.id = id;
        this.permissions = permissions;
    }

    public String getId() {
        return id;
    }

    public List<String> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<String> permissions) {
        this.permissions = permissions;
    }
}

class GeneralUser extends UserPermissions {
    public GeneralUser(String id, List<String> permissions) {
        super(id, permissions);
    }
}

class AdministrativeAssistantUser extends UserPermissions {
    public AdministrativeAssistantUser(String id, List<String> permissions) {
        super(id, permissions);
    }
}

class AdministrativeUser extends UserPermissions {
    public AdministrativeUser(String id, List<String> permissions) {
        super(id, permissions);
    }
}
